micro-deps/micro-deps-spring-config/src/main/java/com/ofg/infrastructure/discovery/ServiceDiscoveryInfrastructureConfiguration.java

```diff
@@ -86,18 +87,34 @@ public ServiceInstance serviceInstance(MicroserviceAddressProvider addressProvid
                 .build();
     }
 
-    @Bean(initMethod = "start", destroyMethod = "close")
+    @Bean(destroyMethod = "close")
     @SuppressWarnings("unchecked")
     public ServiceDiscovery serviceDiscovery(CuratorFramework curatorFramework,
                                              ServiceInstance serviceInstance,
                                              ServiceConfigurationResolver serviceConfigurationResolver) {
         log.info("Registering myself: " + String.valueOf(serviceInstance));
-        return ServiceDiscoveryBuilder
+        final ServiceDiscovery<Void> serviceDiscovery = ServiceDiscoveryBuilder
                 .builder(Void.class)
                 .basePath("/" + serviceConfigurationResolver.getBasePath())
                 .client(curatorFramework)
                 .thisInstance(serviceInstance)
                 .build();
+        asyncRegisterInsideZookeeper(serviceDiscovery);
+        return serviceDiscovery;
+    }
+
+    private void asyncRegisterInsideZookeeper(final ServiceDiscovery<Void> serviceDiscovery) {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    serviceDiscovery.start();
```

Comment:
`start()` is blocking? So please add a log message afterwards that it succeeded. Maybe just move `"Registering myself"` here?